src/Generator/Generators/CLI/CLITypeReferences.cs

```diff
@@ -48,10 +48,27 @@ public CLITypeReference GetTypeReference(Declaration decl)
             if (typeReferences.ContainsKey(decl))
                 return typeReferences[decl];
 
-            var @ref = new CLITypeReference { Declaration = decl };
-            typeReferences.Add(decl, @ref);
+            var translationUnit = decl.Namespace.TranslationUnit;
+
+            if (ShouldIncludeTranslationUnit(translationUnit) && decl.IsGenerated && !IsBuiltinTypedef(decl))
+            {
+                var @ref = new CLITypeReference { Declaration = decl };
+
+                @ref.Include = new CInclude
+                {
+                    File = GetIncludePath(translationUnit),
+                    TranslationUnit = translationUnit,
+                    Kind = translationUnit.IsGenerated
+                    ? CInclude.IncludeKind.Quoted
```

Comment:
This should be indented either one level up for both the lines, or you can keep the `?` in the previous line, and have the rest of the conditional expression on a single line, indented.